packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -144,16 +157,15 @@ export async function generateBlogPosts(
 
       // Use file create time for blog.
       date = date || (await fs.stat(source)).birthtime;
+
+      const slug =
+        frontMatter.slug || (match ? toUrl({date, link: linkName}) : linkName);
       frontMatter.title = frontMatter.title || linkName;
 
       blogPosts.push({
-        id: frontMatter.id || frontMatter.title,
+        id: frontMatter.slug || frontMatter.title,
         metadata: {
-          permalink: normalizeUrl([
-            baseUrl,
-            routeBasePath,
-            frontMatter.id || toUrl({date, link: linkName}),
-          ]),
+          permalink: normalizeUrl([baseUrl, routeBasePath, formatSlug(slug)]),
```

Comment:
For now I'd be more comfortable if you did not format the slug if it's provided by frontmatter.

The docs plugin does not do any normalization either so I prefer it to be consistent.

The user should be able to use `slug: /hey/my super path/héllô` and it should work